packages/swapper/src/api.ts

```diff
@@ -4,9 +4,36 @@ import type { CosmosSdkChainId, EvmChainId, UtxoChainId } from '@shapeshiftoss/c
 import { createErrorClass } from '@shapeshiftoss/errors'
 import type { HDWallet } from '@shapeshiftoss/hdwallet-core'
 import type { ChainSpecific, KnownChainIds, UtxoAccountType } from '@shapeshiftoss/types'
+import type { Result } from '@sniptt/monads'
 
 export const SwapError = createErrorClass('SwapError')
 
+export type SwapErrorRight = {
+  name: 'SwapError'
+  message: string
+  cause?: unknown
+  details?: unknown
+  code?: string
+}
+
+export const makeSwapErrorRight = ({
```

Comment:
We probably want to wrap the return in `Err`, as this seems to be the only consumed use case.

Reply:
Unfortunately we seem to lose type narrowing here by making an `Err` outside of the local context:

<img width="1174" alt="image" src="https://user-images.githubusercontent.com/17035424/232792487-58485775-33f5-4440-b6db-979365318600.png">

Happy to do it as a follow-up since it will be a huge diff and a bit more plumbing to figure out the types shenanigans